### PR TITLE
[client][test] Client multiprocessing tests + client api minor fix

### LINKED ISSUE #16895: [ray client][test] Add client tests for ray.multiprocessing

### PR DESCRIPTION
**Pull request by DmitriGekhtman**
## Why are these changes needed?

Adds `test_multiprocessing` to client tests.
Also fixes client `is_initialized` and `_internal_kv` connected apis (these were causing the tests to fail when run in client mode.)

## Related issue number

Closes https://github.com/ray-project/ray/issues/16895

## Checks

- [x] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [x] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Review by mwtian — APPROVED** [review]

File: python/ray/util/client/__init__.py [hunk 1]
```diff
@@ -143,10 +143,14 @@ def remote(self, *args, **kwargs):
         return self.api.remote(*args, **kwargs)
 
     def __getattr__(self, key: str):
-        if not self.is_connected():
+        if self.is_connected():
+            return getattr(self.api, key)
+        elif key in ["is_initialized", "_internal_kv_initialized"]:
+            # Client is not connected, thus Ray is not considered initialized.
+            return lambda: False
+        else:
```
Comment (ijrsvt) [inline python/ray/util/client/__init__.py:151]: Thanks for this fix!

**Comment by ijrsvt** [comment]
Confirmed that the test did run!

Screen Shot 2021-07-06 at 3 27 45 PM

**Review by ijrsvt — APPROVED** [review]
LGTM!

**Review by ijrsvt — APPROVED** [review]
LGTM!

**Comment by ijrsvt** [comment]
@DmitriGekhtman The test: `test_multiprocessing_client_mode.py` is failing on Windows :(

**Comment by DmitriGekhtman** [comment]
It looks like it timed out after 40-some seconds on windows, took 110-ish seconds on linux.
Need to move it to the large bucket on windows..

**Comment by DmitriGekhtman** [comment]
actually looks like non-client version is skipped, so skipping client one as well

**Comment by DmitriGekhtman** [comment]
@ijrsvt Looks good to merge now.

**Pull request merged**